### PR TITLE
#614 | Leela Prasanth | Adds an endpoint to create consent-request fo…

### PR DESCRIPTION
**Pull request by leelanakka**
…r patients by the patient

### CODE REVIEW

File: src/main/java/in/org/projecteka/hiu/SecurityConfiguration.java [hunk 1]
```diff
@@ -48,7 +48,8 @@
             V_1_HEALTH_INFORMATION_HIU_ON_REQUEST
     };
 
-    private static final List<Map.Entry<HttpMethod, String>> CM_PATIENT_APIS = List.of(Map.entry(HttpMethod.GET, "/cm/hello"));
+    private static final List<Map.Entry<HttpMethod, String>> CM_PATIENT_APIS = List.of(Map.entry(HttpMethod.GET, "/cm/hello")
+            , Map.entry(HttpMethod.POST, "/hiu/patient/consent-request"));
```
Comment (dheeraj-p) [inline src/main/java/in/org/projecteka/hiu/SecurityConfiguration.java]: Can you format it better.

File: src/main/java/in/org/projecteka/hiu/consent/ConsentService.java [hunk 2]
```diff
@@ -87,15 +104,44 @@ public ConsentService(HiuProperties hiuProperties,
...
         return validateConsentRequest(consentRequestData)
-                .then(sendConsentRequestToGateway(requesterId, consentRequestData));
+                .then(sendConsentRequestToGateway(requesterId, consentRequestData, gatewayRequestId));
+    }
+
+    public Mono<Map<String, String>> handlePatientConsentRequest(String requesterId, PatientConsentRequest consentRequest) {
+        Map<String, String> response = new HashMap<>();
+        return Flux.fromIterable(consentRequest.getHipIds())
+                .flatMap(hipId -> buildConsentRequest(requesterId).flatMap(consentRequestData -> {
+                    var patientRequestId = UUID.randomUUID();
```
Comment (dheeraj-p) [inline src/main/java/in/org/projecteka/hiu/consent/ConsentService.java]: Rename this to dataRequestId as we discussed.

File: src/main/java/in/org/projecteka/hiu/consent/ConsentService.java [hunk 3]
```diff
@@ -236,6 +286,15 @@ public String load(String key) {
...
                 });
+//        this.patientRequestCache = CacheBuilder
+//                .newBuilder()
+//                .maximumSize(50)
+//                .expireAfterWrite(10, TimeUnit.MINUTES)
+//                .build(new CacheLoader<>() {
+//                    public String load(String key) {
+//                        return "";
+//                    }
+//                });
```
Comment (dheeraj-p) [inline src/main/java/in/org/projecteka/hiu/consent/ConsentService.java]: Please remove the commented code.

Comment (leelanakka) [reply src/main/java/in/org/projecteka/hiu/consent/ConsentService.java]: done

File: src/main/java/in/org/projecteka/hiu/consent/ConsentService.java (hunk 2 above)
Comment (leelanakka) [reply src/main/java/in/org/projecteka/hiu/consent/ConsentService.java]: done

File: src/main/java/in/org/projecteka/hiu/SecurityConfiguration.java (hunk 1 above)
Comment (leelanakka) [reply src/main/java/in/org/projecteka/hiu/SecurityConfiguration.java]: done

**Review by dheeraj-p — APPROVED** [review]

**Pull request merged**